### PR TITLE
chore(release): 2.0.1 [skip ci]

### PR DESCRIPTION
**Pull request by yongchayut**
**Type of PR** -> Feature

**Ticket** -> https://siampiwat.atlassian.net/browse/SP-7993
**Also impact**
- https://siampiwat.atlassian.net/browse/SP-7994
- https://siampiwat.atlassian.net/browse/SP-7995

### Description

* **feat:** log every perx response with more useful data for better investigation

### Proof of work

Log message from get wallet request via payment service

**Normal Request/Response logging**

Screenshot 2567-07-22 at 22 37 36

**Error without response property**
Screenshot 2567-07-22 at 22 40 41

**Error without both request and response**
Screenshot 2567-07-22 at 22 43 18

### CODE REVIEW

File: CHANGELOG.md
[the remark this answers is not in the document]
Comment (yongchayut) [reply CHANGELOG.md:1]: Okay krub, so we just need to remove the diff ? 
just confused that this diff generated on pre-commit

File: src/client.ts [hunk 1]
```diff
@@ -497,40 +502,50 @@ export class PerxService implements IPerxService {
...
-        this._sendMetric(
-          this.config.baseURL,
-          error.config.url || '',
-          responseTime,
-          error.response.status,
-          'fail',
-        )
-        throw error
+        try {
+          const responseTime = logAxiosResponse(error.response)
```
[the remark this answers is not in the document]
Comment (yongchayut) [reply src/client.ts:516]: Yes, I belief the connection error would be logged here.

File: src/utils/index.ts [hunk 2]
```diff
@@ -1 +1,42 @@
...
+
+    // formatting datetime
+    requestedAtString = new Date(requestedAt).toISOString()
+  }
+
+  const summaryLine = `${status} [${request.method}] – ${fullUrl}`
+  // TODO
+  // const rawRequestHeader = request._header
+
+  Loggy.log(`[PerxClient:AccessLog] ${requestedAtString} : ${summaryLine}`, statusText,
```
Comment (yongchayut) [inline src/utils/index.ts:32]: The request information would also be logged here as well krub

File: src/client.ts [hunk 3]
```diff
@@ -497,40 +502,50 @@ export class PerxService implements IPerxService {
     })
```
[the remark this answers is not in the document]
Comment (yongchayut) [reply src/client.ts:503]: As mention below, the request info is also logged in the `logAxiosResponse` function krub. 
And all errors would be logged in the `(error) => {}` function, let's say network timeout.

File: src/client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yongchayut) [reply src/client.ts]: @eakvelocitytech Thanks for suggestion, added logging for the error that might not have `response` as in the docs -> https://github.com/axios/axios/tree/v0.21.1?tab=readme-ov-file#handling-errors

File: src/client.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (yongchayut) [reply src/client.ts:503]: As we discussed, the request info would be logged with the response info into a single line log format. Thanks for clarify

**Review by eakvelocitytech — APPROVED** [review]

**Pull request merged**